libs/go/libp2p_node/dht/dhtpeer/dhtpeer.go

```diff
@@ -1821,3 +1908,23 @@ func (dhtPeer *DHTPeer) registerAgentAddress(addr string) error {
 	}
 	return nil
 }
+
+func (dhtPeer *DHTPeer) AddACNStatusMessage(status *acn.Status, counterpartyID string) {
+	dhtPeer.acnStatuses[counterpartyID] <- status
```

Comment:
no need of lock? Could not happen `dhtPeer.acnStatuses[counterpartyID]` is deleted?

Reply:
lock added